src/esm.ts

```diff
@@ -72,6 +78,35 @@ export function registerAndCreateEsmHooks(opts?: RegisterOptions) {
     );
   }
 
+  async function load(
+    url: string,
+    context: {},
+    defaultLoad: typeof load
+  ): Promise<{ format: Format; source: string | Buffer }> {
+    const { format } = await getFormat(url, context, defaultGetFormat);
```

Comment:
What if `context.format` has already been set, perhaps because we have been composed with another loader?  Does `defaultGetFormat` respect this?

If I remember correctly, `defaultLoad` will respect an incoming `format` hint and will not overwrite it, so I think we should match that behavior.  Please correct me if I'm wrong.

Reply:
You mean we should check if context already have a `format` and if so skip the call to `getFormat`?

Reply:
I added a check for using existing format hint on the context, let me know if this is what you expected.